client/client.go

```diff
@@ -22,57 +22,318 @@ limitations under the License.
+		err        error
+		workingSet uint64
+	)
+	container := &docker.Container{}
+
+	var pid int
+
+	if !isHost(id) {
+		if !isFullLengthID(id) {
+			return stats, fmt.Errorf("Container id %+v is not fully-length - cannot inspect container", id)
+		}
+		// inspect container based only on fully-length container id.
+		container, err = dc.InspectContainer(id)
+
+		if err != nil {
+			// only log error message and return stats (contains cgroups stats)
+			return stats, nil
+		}
+		pid = container.State.Pid
+	}
+
+	for cg, stat := range groupWrap {
+		var groupPath string
+		var err error
+
+		groupPath, err = GetSubsystemPath(cg, id)
+		if err != nil {
+			continue
+		}
+		// get cgroup stats for given docker
+		err = stat.GetStats(groupPath, stats.CgroupStats)
+		if err != nil {
+			// just log about it
+			if isHost(id) {
+				fmt.Fprintln(os.Stderr, "Cannot obtain cgroups statistics for host, err=", err)
+			} else {
+				fmt.Fprintln(os.Stderr, "Cannot obtain cgroups statistics for container: id=", id, ", image=", container.Image, ", name=", container.Name, ", err=", err)
+			}
+			continue
+		}
+	}
+
+	// calculate additional stats memory:working_set based on memory_stats
+	if totalInactiveAnon, ok := stats.CgroupStats.MemoryStats.Stats["total_inactive_anon"]; ok {
+		workingSet = stats.CgroupStats.MemoryStats.Usage.Usage
+		if workingSet < totalInactiveAnon {
+			workingSet = 0
+		} else {
+			workingSet -= totalInactiveAnon
+		}
+
+		if totalInactiveFile, ok := stats.CgroupStats.MemoryStats.Stats["total_inactive_file"]; ok {
+			if workingSet < totalInactiveFile {
+				workingSet = 0
+			} else {
+				workingSet -= totalInactiveFile
+			}
+		}
+	}
+	stats.CgroupStats.MemoryStats.Stats["working_set"] = workingSet
+
+	if !isHost(id) {
+		rootFs := "/"
+
+		stats.Network, err = network.NetworkStatsFromProc(rootFs, pid)
+
+		if err != nil {
+			// only log error message
+			fmt.Fprintf(os.Stderr, "Unable to get network stats, containerID=%+v, pid %d: %v", container.ID, pid, err)
+		}
+
+		stats.Connection.Tcp, err = network.TcpStatsFromProc(rootFs, pid)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to get tcp stats from pid %d: %v", pid, err)
+		}
+
+		stats.Connection.Tcp6, err = network.Tcp6StatsFromProc(rootFs, pid)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to get tcp6 stats from pid %d: %v", pid, err)
+		}
+
+	} else {
+		stats.Network, err = network.NetworkStatsFromRoot()
+		if err != nil {
+			// only log error message
+			fmt.Fprintf(os.Stderr, "Unable to get network stats, containerID=%v, %v", id, err)
+		}
+
+	}
+	if collectFs {
+		stats.Filesystem, err = fs.GetFsStats(container)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to get filesystem stats for docker: %v, err=%v", id, err)
+		}
+	}
+
+	return stats, nil
+}
+
+// getShortId returns short version of container ID (12 char)
+func GetShortId(dockerID string) (string, error) {
```

Comment:
Consider changing name of this function into "GetShortID", name of the next function in this file is "FullLengthID" so to have consistency I'm suggesting this change.